packages/firestore/src/api/credentials.ts

```diff
@@ -280,9 +282,11 @@ export class FirstPartyToken implements Token {
 export class FirstPartyCredentialsProvider implements CredentialsProvider {
   constructor(private gapi: Gapi, private sessionIndex: string) {
     assert(
-      this.gapi &&
-        this.gapi['auth'] &&
-        this.gapi['auth']['getAuthHeaderValueForFirstParty'],
+      !!(
```

Comment:
Would it make sense to move this assert to `makeCredentialsProvider`? If we do that, we could then change the type of [FirestoreSettings.credentials](https://github.com/firebase/firebase-js-sdk/blob/54bdfc4cebb038ffe19b16bb263b1007132ca356/packages/firestore/src/api/database.ts#L156) to `unknown` and convert it to `Gapi` in  `makeCredentialsProvider()`.

Feel free to punt since this may very well not be worth your time :)

Reply:
Oh!  I like that idea quite a bit.  Done.